### DISCUSSION TITLE
Custom HP threshold for fade playfield to red when health is low

### DISCUSSION THREAD

**Discussion by gagahpangeran** [ppy/osu#13133] [Ideas]
**Describe the new feature:** 

Currently the red overlay is start to fade when HP is very low. It will be nice if user can adjust the HP threshold. The minimum value is 0% (disabled) until 50% of HP.

**Proposal designs of the feature:** 

Slider in setting from 0 to 50, value 0 for disabled it.

**Comment by bdach** [comment]
Personally disagree with having this as a setting on principle of keeping setting count as low as possible.

**Comment by Game4all** [comment]
It doesn't really make sense (at least to me) to have the ability to set the health threshold at which the layer should start fading in. Since the threshold is hardcoded as a constant
https://github.com/ppy/osu/blob/fb5a54d2422171d8f2cb40ee1cb4b9b96dabc4f7/osu.Game/Screens/Play/HUD/FailingLayer.cs#L37 
maybe it should just be changed to something a bit more higher if it was deemed too low?

**Comment by gagahpangeran** [comment]
> It doesn't really make sense (at least to me) to have the ability to set the health threshold at which the layer should start fading in. Since the threshold is hardcoded as a constant
> 
> https://github.com/ppy/osu/blob/fb5a54d2422171d8f2cb40ee1cb4b9b96dabc4f7/osu.Game/Screens/Play/HUD/FailingLayer.cs#L37
> 
> maybe it should just be changed to something a bit more higher if it was deemed too low?

That constant can be changed to variable and the value is from settings.
